### ISSUE TITLE
Feature Request: Mail Rules by Date

### ISSUE THREAD

**Issue by jsonbecker** [nylas/N1#3098]
Mail rules currently have options for operating on from, to, cc, and other meta data in addition to the subject and body text. I would like to create rules based on dates-- for example, messages that are more than thirty days old or that have a certain subject during a certain date range.

**Comment by manicmarvin** [comment]
You can use the same gmail syntax to search by date. For example: `after:2017/1/3 before:2017/1/4 ` will return all mails that arrived in the selected box (like your Inbox). You can also search a specific label/folder with `after:2017/1/3 before:2017/1/4 in:BQFF` _(where BQFF is a label I use)_

**Comment by OJFord** [comment]
@manicmarvin Please correct me if I've misunderstood, but that's for searching; not mail rules?

Also, that's not Gmail's syntax? Or if it is, it has aliases which I use:
- `date-begin`, `date-end`
- `label`

These don't work in N1. Thanks though, I've been missing that!

**Comment by manicmarvin** [comment]
@OJFord You're absolutely right, it is only for search and not mail rules. I hadn't realised, because I use teh google web mail interfact to set up filters and rules, which I find are faster anyway. 

(Unsure about the date aliases you're using, but `label: <label-name>` does work for me.)

But this would definetely be a useful feature @spang.

**Comment by OJFord** [comment]
Unless you're setting the mail rule to run "until" some time in the future, they'll either never or always apply to incoming mail - and only be used mail rules for the 'process all' button.

With that in mind, this is related to/part of #3165 - in which I'm suggesting a way to run one-off (possibly saved) jobs for more advanced/controlled management.

**Comment by manicmarvin** [comment]
Yes, but I imagine OP ( @jsonbecker ) intends to use the rule to bulk process historic mail of a certain date range. Using 'process all'.

Your idea #3165 sounds fantastic, but I think this request is a seperate fix. I also imagine this will easier and faster to implement.

**Comment by jsonbecker** [comment]
@manicmarvin the use case I was thinking of, that I think is faster and simpler (although less awesome) than #3165 is basically a live rule that says "any mail older than 30 days mark as read and archive". Or "anything that is in the inbox and is 7 days old flat as important". It's most useful in a "process all" scenario, sure, but It'd be nice in a more live context. Of course, if Mail Rules are only ever checked against incoming mail and not checked against the entire box, then I guess my case only would work for "process all"

**Comment by OJFord** [comment]
Ooh, I like that too. I didn't realise you meant relative dates.

(Obviously, [I can't read](https://github.com/nylas/N1/issues/3098#issue-191913490)...)
